### PR TITLE
Contact Form: Toolbar Improvements

### LINKED ISSUE #27029: Form Block Toolbar: Redesign Data Management / Submissions Settings incl. Icon
Image

If you have selected the form parent-block, you have a "pen" icon in the toolbar. This shows you the same settings area as in the sidebar. I am not sure if the icon is expressive enough, and combining data management with the behavior of the block after submission is confusing.

General question: Should the "settings editing" happen there in the dropdown, or should there be just a link to where you can edit the details of the settings? Especially relevant, as we are adding many more data connection options soon.

The dropdown can focus on communicating all the data management options and listing them.

pcsBup-kr-p2

### PR DESCRIPTION
**Pull request by jcheringer**
Fixes #27029

#### Changes proposed in this Pull Request:

* Include a Toolbar dropdown that allows managing some Forms settings

image

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Open/create a Post/Page and add a `Form` block
* Select the Form block and check if the new button is present in the Toolbar
* Play with the settings to ensure they work as expected

### CODE REVIEW

**Comment by digitalwaveride** [comment]
João, looks great! 👍  Really fun to play with it!

From a design perspective I noticed a few little things with the modal styling that could be finetuned:

If you compare the "Transform to" modal with the "Connect to" modal, you can see little differences:

image

[ ]  One is the outline color of the modal, 

[ ]  the other is the styling and padding of the "all caps headline".

both please adjust to the "Transform to" state.

 [ ] A third little thing is the cursor state over the "X" which is not a "pointer" at the moment.

Thank you!

File: projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-form-settings-dropdown.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		);
+	}
+
+	return (
+		<div style={ { padding: '12px' } }>
+			<Flex style={ { marginBottom: '24px' } }>
+				<FlexItem>
+					<strong>{ currentSettingsView.title }</strong>
+				</FlexItem>
+				<Icon icon={ close } onClick={ () => setCurrentSettingsView( null ) } size={ 16 } />
```
Comment (ice9js) [inline projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-form-settings-dropdown.js]: Nit: should we wrap this in a `<button>` for accessibility reasons?

**Review by ice9js — DISMISSED** [review]
Works for me - I just left a small comment regarding the `x` button on the popover. At least, let's give it `cursor: pointer;`?

File: projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-form-settings-dropdown.js (hunk 1 above)
Comment (jcheringer) [reply projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-form-settings-dropdown.js]: ✅

**Comment by jcheringer** [comment]
Thanks, @digitalwaveride, and @ice9js, for the review.
I addressed the suggestions in the last commit.

**Review by ice9js — DISMISSED** [review]

**Review by samiff — APPROVED** [review]
Very nice! All dropdown actions worked for me, and updates were reflected in the corresponding sidebar inputs.

**Comment by oskosk** [comment]
We need to address this icon @digitalwaveride. This icon is used for representing "plugin" in other contexts

**Pull request merged**